frame/support/src/debug.rs

```diff
@@ -170,8 +170,16 @@ impl RuntimeLogger {
 	/// This is a no-op when running natively (`std`).
 	#[cfg(not(feature = "std"))]
 	pub fn init() {
-		static LOGGER: RuntimeLogger = RuntimeLogger;;
+		static LOGGER: RuntimeLogger = RuntimeLogger;
 		let _ = log::set_logger(&LOGGER);
+
+		// Set max level to `TRACE` to ensure we propagate
+		// all log entries to the native side that will do the
+		// final filtering on what should be printed.
+		//
+		// If we don't set any level, logging is disabled
+		// completly.
+		log::set_max_level(log::LevelFilter::Trace);
```

Comment:
AFAIR it shouldn't matter. We already return `true` in `log::Log::enabled` which should be responsible for filtering like that? Are you sure it's actually filtered somewhere internally in `log` crate?

Reply:
Yes. I checked that. Remove the line and the test fails ;)

Reply:
https://docs.rs/log/0.4.11/log/#implementing-a-logger

They also mention it here that you need to call it.

Reply:
https://docs.rs/log/0.4.11/src/log/macros.rs.html#46

Here they call the max_level function that checks the value that is set with set_max_value

Reply:
Fair enough!